src/Uno.UI/UI/Xaml/UIElement.Pointers.cs

```diff
@@ -1071,6 +1075,38 @@ internal bool OnPointerUp(PointerRoutedEventArgs args, BubblingContext ctx = def
 			return handledInManaged;
 		}
 
+		private void UpdateRaisedEventFlags(PointerRoutedEventArgs args)
+		{
+			if (!IsGestureRecognizerCreated)
+			{
+				return;
+			}
+
+			var pointerId = args.GetCurrentPoint(this).PointerId;
+			var eventsRaised = args.EventsAlreadyRaised;
+			if (eventsRaised.HasFlag(GestureOrientedPointerEvents.Tap))
+			{
+				GestureRecognizer.PreventTap(pointerId);
+			}
+
+			if (eventsRaised.HasFlag(GestureOrientedPointerEvents.RightTap))
+			{
+				GestureRecognizer.PreventRightTap(pointerId);
+			}
+
+			if (eventsRaised.HasFlag(GestureOrientedPointerEvents.DoubleTap))
+			{
+				GestureRecognizer.PreventDoubleTap(pointerId);
+			}
+
+			if (eventsRaised.HasFlag(GestureOrientedPointerEvents.Hold))
+			{
+				GestureRecognizer.PreventHolding(pointerId);
+			}
```

Comment:
Even if it what I suggested, I realized that here we have 4 dictionary lookup to get the current `_gesture` for the given `pointerId` in the `GestureRecognizer`. I would instead move this in the recognizer and add a method `Prevent(uint pointerId, GestureOrientedPointerEvents events)`. This method could even return the "still active event", i.e. what is returned by the `CanRaiseGestureOrientedEvents(pointerId)`

Reply:
I opted to return the updated inactive events directly (i.e. events | still active). Minor difference, but feels smoother. Let me know if you don't like it.

Reply:
I changed my mind. Keeping it generic and not specific to "already raised events" makes more sense.